packages/modal/src/ModalView.tsx

```diff
@@ -0,0 +1,300 @@
+  transition: all 150ms ease-in-out;
+  margin: auto;
+  max-height: calc(100% - ${defaultVerticalSpacing}px);
+  position: relative;
+  pointer-events: all;
+  transform: translate3d(0, -16px, 0);
+  opacity: 0;
+
+  &:focus {
+    outline: none;
+  }
+`;
+
+const modeStyles = css`
+  color: ${uiColors.gray.dark3};
+  background-color: ${uiColors.white};
+  font-family: ${fontFamilies.default};
+  border-radius: 24px;
+  padding: 35px 40px;
+  box-shadow: 0px 8px 20px -8px ${transparentize(0.4, palette.black)};
+`;
+
+const visibleModalContentStyle = css`
+  transform: translate3d(0, 0, 0);
+  opacity: 1;
+`;
+
+const modalSizes: { readonly [K in ModalSize]: string } = {
+  small: css`
+    width: 400px;
+  `,
+
+  default: css`
+    width: 600px;
+  `,
+
+  large: css`
+    ${mq({
+      width: ['720px', '720px', '960px'],
+    })}
+  `,
+};
+
+const baseCloseButtonStyles = css`
+  position: absolute;
+  cursor: pointer;
+`;
+
+const closeButton: Record<
+  Mode,
+  Record<CloseIconColor, string> & Record<'position', string>
+> = {
+  [Mode.Light]: {
+    [CloseIconColor.Default]: css`
+      color: ${palette.gray.base};
+    `,
+    [CloseIconColor.Dark]: css`
+      color: ${palette.gray.dark1};
+    `,
+    [CloseIconColor.Light]: css`
+      color: ${palette.white};
+    `,
+    position: css`
+      // x-icon should be 24px from edge. IconButton is 28x28 and Icon is 16x16
+      // so there's already (28 - 16) / 2 = 6px of spacing. 24 - 6 = 18.
+      right: 18px;
+      top: 18px;
+    `,
+  },
+  [Mode.Dark]: {
+    [CloseIconColor.Default]: css`
+      color: ${uiColors.gray.base};
+
+      &:hover {
+        color: ${uiColors.gray.base};
+      }
+    `,
+    [CloseIconColor.Dark]: css`
+      color: ${uiColors.gray.base};
+
+      &:hover {
+        color: ${uiColors.gray.base};
+      }
+    `,
+    [CloseIconColor.Light]: css`
+      color: ${uiColors.gray.base};
+
+      &:hover {
+        color: ${uiColors.gray.base};
+      }
+    `,
+    position: css`
+      // x-icon should be 16px from edge. IconButton is 28x28 and Icon is 16x16
+      // so there's already (28 - 16) / 2 = 6px of spacing. 16 - 6 = 10.
+      right: 10px;
+      top: 10px;
+    `,
+  },
+};
+
+function ModalView({
+  open = false,
+  size = ModalSize.Default,
+  setOpen = () => {},
+  shouldClose = () => true,
+  darkMode = false,
+  children,
+  className,
+  contentClassName,
+  initialFocus,
+  closeIconColor = CloseIconColor.Default,
+  ...rest
+}: ModalProps) {
+  const mode = darkMode ? Mode.Dark : Mode.Light;
+
+  const nodeRef = React.useRef(null);
+
+  const [modalRef, setModalRef] = useState<null | HTMLDivElement>(null);
+
+  const { isPopoverOpen } = usePopoverContext();
+
+  const handleClose = useCallback(() => {
+    if (setOpen && shouldClose()) {
+      setOpen(false);
+    }
+  }, [setOpen, shouldClose]);
+
+  const id = useIdAllocator({ prefix: 'modal' });
+
+  useEscapeKey(handleClose, { enabled: open && !isPopoverOpen });
```

Comment:
Hard to tell but this is what I updated, If the popover is not open then we can close the modal